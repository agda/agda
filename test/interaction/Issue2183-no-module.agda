-- Andreas, 2017-03-27, issue #2183
-- Better error message for splitting on non-visible dot pattern.
-- Andreas, 2025-10-12, issue #7941
-- Allow splitting on non-visible dot pattern.

open import Agda.Builtin.Equality
open import Agda.Builtin.Nat

data Fin : Nat → Set where
  fzero : ∀ n → Fin (suc n)

test : ∀ n (i : Fin n) → Set
test n (fzero m) = {!n!}  -- C-c C-c

-- WAS error:
-- Cannot split on variable n, because it is bound to term  suc m
-- when checking that the expression ? has type Set

-- Behavior since 2025-10-12: succeed and give
-- test .(suc m) (fzero m) = {!!}
