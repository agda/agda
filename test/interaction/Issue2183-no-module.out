(agda2-status-action "")
(agda2-info-action "*Type-checking*" "" nil)
(agda2-highlight-clear)
(agda2-status-action "")
(agda2-info-action "*All Goals*" "?0 : Set " nil)
((last . 1) . (agda2-goals-action '(0)))
(agda2-info-action "*Error*" "Issue2183-no-module.agda:11,21-26: error: [Interaction.CaseSplitError] Cannot split on variable n, because it is bound to suc m when checking that the expression ? has type Set" nil)
((last . 3) . (agda2-maybe-goto '("Issue2183-no-module.agda" . 273)))
(agda2-highlight-load-and-delete-action)
(agda2-status-action "")
